search_engine_parser/core/engines/google.py

```diff
@@ -24,7 +24,9 @@ def parse_soup(self, soup):
         # find all class_='g' => each result
         return soup.find_all('div', class_='g')
 
-    def parse_single_result(self, single_result):
+    # Elements is a 2d array consting of names, html tags and classes
+    # Elements = [['link_tags', 'a', 'r'], ['r_elem', 'div', 'st'] ... ]
+    def parse_single_result(self, single_result, element):
```

Comment:
the `element` args should be changed to a keyword `mode` whose default is `full`. Using this `mode`, you can then choose to return process appropriately. This reduces the complexities of having to pass element definition everytime this function should be called. All logic about an engine should be in the engine and not passed as extra args.